r/R/dplyr-functions.R

```diff
@@ -108,6 +108,10 @@ nse_funcs$is.infinite <- function(x) {
   is_inf & !nse_funcs$is.na(is_inf)
 }
 
+nse_funcs$abs <- function(x){
+  Expression$create("abs_checked", x)
```

Comment:
I used `abs_checked` to be in sync with [ARROW-12575: [R] Use unary negative kernel](https://github.com/apache/arrow/pull/10196), see [discussion](https://github.com/apache/arrow/pull/10196/files/288e06ed681c170de587591b88c6b37f96a3eddd#r625885713).